src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,49 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int [] diagonal = new int [matrix.length];
+        for (int i = 0;i < matrix.length;i++) {
+            for (int j = 0;j < matrix.length;j++) {
+                if (i == j) {
+                    diagonal[i] = matrix[i][j];
+                }
+            }
+        }
+        return diagonal;
+
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] counterDiagonal = new int [matrix.length];
+        for (int i = 0;i < matrix.length;i++) {
```

Comment:
```suggestion
        for (int i = 0; i < matrix.length; i++) {
```
please pay attention to a whitespaces

Reply:
done